server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -3181,6 +3184,13 @@ private Pair<List<DiskOfferingJoinVO>, Integer> searchForDiskOfferingsInternal(L
             sc.addAnd("encrypt", SearchCriteria.Op.EQ, encrypt);
         }
 
+        if (storageType != null) {
+            Boolean isLocalStorageType = getStorageType(storageType);
+            if (isLocalStorageType != null) {
+                sc.addAnd("useLocalStorage", Op.EQ, isLocalStorageType);
+            }
+        }
+
```

Comment:
```
        if (storageType != null) {
            Boolean isLocalStorageType = getStorageType(storageType);
            if (isLocalStorageType != null) {
                sc.addAnd("useLocalStorage", Op.EQ, isLocalStorageType);
            }
        }
```
this is added twice, can you extrract it to a method?

Reply:
I think the code will be more complex if I try to get the class type of the SerachCriteria (sc) generic parameter.